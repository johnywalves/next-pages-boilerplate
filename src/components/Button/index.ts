import styled from 'styled-components'

export const Button = styled.button`
  background-color: transparent;
  color: var(--color-text);
  border: var(--2px) solid var(--color-primary);
  border-radius: var(--4px);
  padding: var(--16px) var(--24px);
  font-size: 1.25rem;
  font-weight: 600;
  cursor: pointer;
  transition: background-color 0.25s, color 0.5s;

  &:hover,
  &:focus {
    background-color: var(--color-primary);
    color: var(--color-background);
  }
`

export default Button
