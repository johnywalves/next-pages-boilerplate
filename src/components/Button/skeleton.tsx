import { ReactNode } from 'react'

import Button from '.'

const ButtonSkeleton = ({ children }: { children?: ReactNode }) => (
  <Button className="skeleton">{children || ' '}</Button>
)

export default ButtonSkeleton
