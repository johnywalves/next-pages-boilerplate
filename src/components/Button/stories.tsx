import { Meta, StoryObj } from '@storybook/react'

import Button, { ButtonSkeleton } from '.'

const meta: Meta<typeof Button> = {
  title: 'Button',
  component: Button,
  tags: ['autodocs'],
  render: ({ children, ...rest }) => (
    <Button {...rest}>
      <p>{children}</p>
    </Button>
  ),
  args: {
    children: 'Texto do bot√£o'
  }
}

type Story = StoryObj<typeof Button>

export const Primary: Story = {
  args: {}
}

export const Secondary: Story = {
  args: {
    variant: 'secondary'
  }
}

export const Tertiary: Story = {
  args: {
    variant: 'tertiary'
  }
}

export const Skeleton: Story = {
  render: () => <ButtonSkeleton />,
  args: {}
}

export default meta
