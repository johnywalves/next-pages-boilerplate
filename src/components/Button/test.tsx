import { render, screen } from '@testing-library/react'

import Button from '.'

const content = 'Texto do botÃ£o'

const Component = () => (
  <Button>
    <p role="text">{content}</p>
  </Button>
)

describe('<Button />', () => {
  it('should render the button and content', () => {
    const { container } = render(<Component />)

    expect(screen.getByRole('button')).toBeInTheDocument()

    expect(screen.getByRole('text')).toBeInTheDocument()

    expect(screen.getByText(content)).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    render(<Component />)

    const button = screen.getByRole('button')

    expect(button).toHaveStyleRule('color', 'var(--color-neutral-100)')

    expect(button).toHaveStyleRule(
      'background-color',
      'var(--color-primary-pure)'
    )
  })
})
