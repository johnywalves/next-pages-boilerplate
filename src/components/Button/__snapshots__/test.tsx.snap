// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the button and content 1`] = `
.c0 {
  background-color: transparent;
  color: var(--color-neutral-100);
  border: var(--2px) solid var(--color-primary);
  border-radius: var(--16px);
  padding: var(--16px) var(--24px);
  font-size: 1.25rem;
  font-weight: 600;
  cursor: pointer;
  transition: background-color 0.25s,color 0.5s;
  display: flex;
  justify-content: center;
  align-items: center;
}

.c0 p {
  display: flex;
  justify-content: center;
  align-items: center;
  vertical-align: middle;
}

.c0:hover,
.c0:focus {
  color: var(--color-neutral-900);
}

.c0.skeleton {
  min-width: 8rem;
}

.c0:hover,
.c0:focus {
  background-color: var(--color-primary);
}

<div>
  <button
    class="c0"
  >
    <p
      role="text"
    >
      Texto do bot√£o
    </p>
  </button>
</div>
`;
